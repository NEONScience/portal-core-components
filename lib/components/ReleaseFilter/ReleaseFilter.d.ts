export default WrappedReleaseFilter;
declare const WrappedReleaseFilter: any;
