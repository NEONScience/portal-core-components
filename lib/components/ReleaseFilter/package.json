{
  "private": true,
  "name": "release-filter",
  "main": "./ReleaseFilter.jsx",
  "module": "./ReleaseFilter.js"
}
