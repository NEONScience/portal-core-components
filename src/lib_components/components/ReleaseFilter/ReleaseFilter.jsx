import React from 'react';
import PropTypes from 'prop-types';

import moment from 'moment';

import { useId } from 'react-id-generator';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import MenuItem from '@material-ui/core/MenuItem';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import Select from '@material-ui/core/Select';
import Typography from '@material-ui/core/Typography';

import Skeleton from '@material-ui/lab/Skeleton';

import CopyIcon from '@material-ui/icons/Assignment';

import Theme from '../Theme/Theme';

const useStyles = makeStyles(theme => ({
  title: {
    fontWeight: 500,
    marginBottom: theme.spacing(1),
  },
  selectInput: {
    width: '100%',
    marginBottom: theme.spacing(0.5),
    backgroundColor: '#fff',
  },
  descriptionContainer: {
    marginTop: theme.spacing(0.5),
  },
  descriptionFlexInnerContainer: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  description: {
    display: 'block',
    color: theme.palette.grey[400],
    overflowWrap: 'break-word',
  },
  descriptionLabel: {
    fontWeight: 700,
    color: theme.palette.grey[400],
    marginRight: theme.spacing(1),
  },
  copyButton: {
    marginTop: theme.spacing(1),
    backgroundColor: '#fff',
    '& svg': {
      width: '0.9rem',
      height: '0.9rem',
    },
  },
  copyButtonAdornment: {
    padding: Theme.spacing(1.25, 1),
    backgroundColor: '#fff',
    marginRight: Theme.spacing(-1.75),
    '& svg': {
      width: '0.9rem',
      height: '0.9rem',
    },
  },
  menuItemSubtitle: {
    color: theme.palette.grey[400],
  },
  horizontalFlex: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  horizontalDescriptions: {
    marginLeft: Theme.spacing(3),
    '& > div:first-child': {
      marginTop: '-2px !important',
    },
  },
}));

const UNSPECIFIED_NAME = 'Latest and Provisional';
const UNSPECIFIED_DESCRIPTION = 'Data in the latest release in addition to provisional data (not yet in any release)';
const DOI_TITLE = 'Digital Object Identifier (DOI) - A citable permanent link to this this data product release';

const formatGenerationDate = (generationDate) => {
  const generationMoment = moment(generationDate);
  return generationMoment ? generationMoment.format('MMMM D, YYYY') : null;
};

const ReleaseFilter = (props) => {
  const classes = useStyles(Theme);
  const {
    excludeNullRelease,
    horizontal,
    maxWidth,
    nullReleaseProductCount,
    onChange,
    releases: releasesProp,
    selected,
    showDoi,
    showGenerationDate,
    showProductCount,
    skeleton,
    title,
    ...otherProps
  } = props;

  const [instanceId] = useId();
  const inputId = `release-filter-input-${instanceId}`;
  const labelId = `release-filter-label-${instanceId}`;

  const releases = [...releasesProp || []].sort(
    (a, b) => (a.generationDate > b.generationDate ? -1 : 1),
  );

  // SANITY CHECK: Render nothing if there are no releases and null release is excluded
  const optionCount = releases.length + (excludeNullRelease ? 0 : 1);
  if (!optionCount) { return null; }

  const findReleaseObject = release => releases.find(entry => entry.release === release) || null;
  const findRelease = release => (findReleaseObject(release) || {}).release || null;
  const getProductCount = (release) => {
    if (!release) { return null; }
    if (Array.isArray(release.dataProducts)) {
      return release.dataProducts.length;
    }
    if (Array.isArray(release.dataProductCodes)) {
      return release.dataProductCodes.length;
    }
    return null;
  };

  let selectedRelease = findRelease(selected);
  if (!selectedRelease && excludeNullRelease) { selectedRelease = releases[0].release; }
  const selectedReleaseObject = findReleaseObject(selectedRelease);

  const handleChange = (newRelease) => {
    const validatedNewRelease = newRelease === UNSPECIFIED_NAME ? null : findRelease(newRelease);
    if (validatedNewRelease === selectedRelease) { return; }
    onChange(validatedNewRelease);
  };

  const maxWidthStyle = maxWidth ? { maxWidth: `${maxWidth}px` } : {};

  const input = (
    <OutlinedInput
      id={inputId}
      name={inputId}
      margin="dense"
      className={classes.selectInput}
      style={maxWidthStyle}
    />
  );

  const titleNode = !title ? null : (
    <Typography variant="h5" component="h3" className={classes.title} id={labelId}>
      {title}
    </Typography>
  );

  // Render skeleton
  if (skeleton) {
    const skeletonStyle = { marginBottom: Theme.spacing(1) };
    return (
      <div {...(otherProps || {})} style={{ maxWidth: `${maxWidth}px`, overflow: 'hidden' }}>
        {titleNode}
        <Skeleton variant="rect" width={maxWidth} height={36} style={skeletonStyle} />
        <Skeleton width="70%" height={16} style={skeletonStyle} />
      </div>
    );
  }

  // Unspecified Node
  const unspecifiedNode = selectedRelease !== null ? null : (
    <div className={classes.descriptionContainer}>
      <Typography variant="caption" className={classes.description}>
        {UNSPECIFIED_DESCRIPTION}
      </Typography>
    </div>
  );

  // Generation Date Node
  let generationDateNode = null;
  if (showGenerationDate && selectedRelease !== null) {
    generationDateNode = (
      <div className={classes.descriptionContainer}>
        <Typography variant="caption" className={classes.description}>
          {`Generated: ${formatGenerationDate(selectedReleaseObject.generationDate)}`}
        </Typography>
      </div>
    );
  }

  // Product Count Node
  let productCountNode = null;
  if (showProductCount) {
    if (selectedRelease === null) {
      productCountNode = nullReleaseProductCount === null ? null : (
        <div className={classes.descriptionContainer}>
          <Typography variant="caption" className={classes.description}>
            {`${nullReleaseProductCount} data products`}
          </Typography>
        </div>
      );
    } else {
      const productCount = getProductCount(selectedReleaseObject);
      productCountNode = (
        <div className={classes.descriptionContainer}>
          <Typography variant="caption" className={classes.description}>
            {productCount !== null ? `${productCount} data products` : 'Unknown data product count'}
          </Typography>
        </div>
      );
    }
  }

  // DOI Node
  let doiNode = null;
  if (showDoi && selectedRelease !== null) {
    doiNode = (
      <div className={classes.descriptionContainer}>
        <div className={classes.descriptionFlexInnerContainer}>
          <Typography
            variant="caption"
            title={DOI_TITLE}
            className={classes.description}
            style={{ marginRight: '4px' }}
          >
            DOI:
          </Typography>
          <Typography
            variant="caption"
            aria-label="doi"
            className={classes.description}
            style={{ overflowWrap: 'anywhere' }}
          >
            {selectedReleaseObject.url}
          </Typography>
        </div>
        <CopyToClipboard text={selectedReleaseObject.url}>
          <Button
            color="primary"
            variant="outlined"
            size="small"
            className={classes.copyButton}
            title={`Copy DOI: ${selectedReleaseObject.url}`}
          >
            <CopyIcon fontSize="small" style={{ marginRight: Theme.spacing(1) }} />
            Copy DOI
          </Button>
        </CopyToClipboard>
      </div>
    );
  }

  // Select Node
  const menuItemSubtitleProps = {
    display: 'block',
    variant: 'caption',
    className: classes.menuItemSubtitle,
  };
  const selectNode = (
    <Select
      data-selenium="release-filter"
      value={selectedRelease || UNSPECIFIED_NAME}
      onChange={event => handleChange(event.target.value)}
      input={input}
      aria-labelledby={labelId}
      renderValue={value => value}
      disabled={optionCount < 2}
    >
      {excludeNullRelease ? null : (
        <MenuItem value={UNSPECIFIED_NAME}>
          <div>
            <Typography display="block">
              {UNSPECIFIED_NAME}
            </Typography>
            <Typography {...menuItemSubtitleProps}>
              {UNSPECIFIED_DESCRIPTION}
            </Typography>
            {!showProductCount || nullReleaseProductCount === null ? null : (
              <Typography {...menuItemSubtitleProps}>
                {`${nullReleaseProductCount} data products`}
              </Typography>
            )}
          </div>
        </MenuItem>
      )}
      {releases.map((entry) => {
        const { release, generationDate } = entry;
        const productCount = getProductCount(entry);
        const productCountSubtitle = productCount !== null
          ? `${productCount} data products`
          : 'Unknown data product count';
        return (
          <MenuItem key={release} value={release}>
            <div>
              <Typography display="block">
                {release}
              </Typography>
              {!showGenerationDate ? null : (
                <Typography {...menuItemSubtitleProps}>
                  {`Generated: ${formatGenerationDate(generationDate)}`}
                </Typography>
              )}
              {!showProductCount ? null : (
                <Typography {...menuItemSubtitleProps}>
                  {productCountSubtitle}
                </Typography>
              )}
            </div>
          </MenuItem>
        );
      })}
    </Select>
  );

  // Final Render
  return horizontal ? (
    <div {...otherProps}>
      <div>
        {titleNode}
      </div>
      <div className={classes.horizontalFlex}>
        <div style={maxWidth ? { width: `${maxWidth}px` } : {}}>
          {selectNode}
        </div>
        <div className={classes.horizontalDescriptions}>
          {unspecifiedNode}
          {generationDateNode}
          {productCountNode}
          {doiNode}
        </div>
      </div>
    </div>
  ) : (
    <div {...otherProps} style={{ width: '100%', ...maxWidthStyle }}>
      {titleNode}
      {selectNode}
      {unspecifiedNode}
      {generationDateNode}
      {productCountNode}
      {doiNode}
    </div>
  );
};

ReleaseFilter.propTypes = {
  excludeNullRelease: PropTypes.bool,
  horizontal: PropTypes.bool,
  maxWidth: PropTypes.number,
  nullReleaseProductCount: PropTypes.number,
  onChange: PropTypes.func,
  releases: PropTypes.arrayOf(
    PropTypes.shape({
      release: PropTypes.string.isRequired,
      generationDate: PropTypes.string.isRequired,
      url: PropTypes.string,
    }),
  ),
  selected: PropTypes.string,
  showDoi: PropTypes.bool,
  showGenerationDate: PropTypes.bool,
  showProductCount: PropTypes.bool,
  skeleton: PropTypes.bool,
  title: PropTypes.string,
};

ReleaseFilter.defaultProps = {
  excludeNullRelease: false,
  horizontal: false,
  maxWidth: 236,
  nullReleaseProductCount: null,
  onChange: () => {},
  releases: [],
  selected: null,
  showDoi: false,
  showGenerationDate: false,
  showProductCount: false,
  skeleton: false,
  title: 'Release',
};

const WrappedReleaseFilter = Theme.getWrappedComponent(ReleaseFilter);

export default WrappedReleaseFilter;
