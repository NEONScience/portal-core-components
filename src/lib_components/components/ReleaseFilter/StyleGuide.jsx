/* eslint react/jsx-one-expression-per-line: 0 */
import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import Divider from '@material-ui/core/Divider';
import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';

import DocBlock from '../../../components/DocBlock';
import CodeBlock from '../../../components/CodeBlock';
import ExampleBlock from '../../../components/ExampleBlock';
import PropsTable from '../../../components/PropsTable';

import ReleaseFilter from './ReleaseFilter';
import Theme from '../Theme/Theme';

const useStyles = makeStyles((theme) => ({
  divider: {
    margin: theme.spacing(3, 0),
  },
}));

const propRows = [
  {
    name: 'excludeNullRelease',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to exclude the n/a option (no release or null release) which is usually
        automatically inserted into the sorted set of release options.
      </p>
    ),
  },
  {
    name: 'horizontal',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to render the subtitle / descriptive elements horizontally to the right of the
        select instead of below.
      </p>
    ),
  },
  {
    name: 'maxWidth',
    type: 'number',
    default: '236',
    examples: (
      <div>
        <tt>250</tt>
        <br />
        <tt>400</tt>
      </div>
    ),
    description: (
      <p>
        CSS maxWidth value to apply to the the select element and its container. Note the default
        value is <tt>236</tt> as this lines up with the inner content width of the sidebar for
        the <Link href="#NeonPage">NeonPage</Link> component, a common location where ReleaseFilter
        may be used.
      </p>
    ),
  },
  {
    name: 'nullReleaseProductCount',
    type: 'number',
    default: 'null',
    examples: (
      <div>
        <tt>100</tt>
        <br />
        <tt>400</tt>
      </div>
    ),
    description: (
      <p>
        The product count value for the null release when <tt>showProductCounts</tt> is true.
        Typically this would represent the number of data products in the n/a release category, or
        the number of data product in the latest release in addition to any providional-only
        data products.
      </p>
    ),
  },
  {
    name: 'onChange',
    type: 'function',
    default: '() => {}',
    description: (
      <p>
        Function to fire when the ReleaseFilter is updated by user interaction. It should take a
        single string argument representing the selected release name (or <tt>null</tt> if n/a
        was selected).
      </p>
    ),
  },
  {
    name: 'releases',
    type: 'array of release objects',
    default: '[]',
    examples: (
      <i>See functional example code below</i>
    ),
    description: (
      <>
        <p>
          Array of objects containing the following keys:
        </p>
        <ul>
          <li><tt>release</tt> (string, required)</li>
          <li><tt>generationDate</tt> (ISO date string, required)</li>
          <li><tt>url</tt> (DOI string, optional)</li>
        </ul>
      </>
    ),
  },
  {
    name: 'selected',
    type: 'string',
    examples: (
      <div>
        <tt>&quot;test-tag-1&quot;</tt>
        <br />
        <tt>null</tt>
      </div>
    ),
    description: (
      <p>
        Initial selected value expressed as a release string or <tt>null</tt>. Note:
        if <tt>excludeNullRelease</tt> is <tt>true</tt> and this prop is <tt>null</tt> or omitted
        then the latest release will automatically be selected.
      </p>
    ),
  },
  {
    name: 'showDoi',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to show the selected release DOI with a copy button below the select input.
      </p>
    ),
  },
  {
    name: 'showGenerationDate',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to show the release generation dates on all menu items and below the select input
        for the selected release. Will not show in either location for the null release.
      </p>
    ),
  },
  {
    name: 'showProductCount',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to show product counts on all menu items and below the select input for the selected
        release. Will not show for the null release if <tt>nullReleaseProductCount</tt> is not also
        provided. In order to work all objects in the <tt>releases</tt> array must contain either
        a <tt>dataProducts</tt> array or a <tt>dataProductCodes</tt> array, the length of which is
        used as a product count. Of those two arrays <tt>dataProducts</tt> takes precedence.
      </p>
    ),
  },
  {
    name: 'showReleaseLink',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to show a link to the release details page for the selected release.
      </p>
    ),
  },
  {
    name: 'releaseLinkDisplayType',
    type: 'one of [Link, Button, ListItem]',
    default: 'Button',
    description: (
      <p>
        When set to show the release link, sets the display type.
      </p>
    ),
  },
  {
    name: 'skeleton',
    type: 'boolean',
    default: 'false',
    description: (
      <p>
        Whether to render a skeleton version of the ReleaseFilter (useful when in a loading state).
      </p>
    ),
  },
  {
    name: 'title',
    type: 'string',
    default: 'Release',
    examples: (
      <div>
        <tt>&quot;Select a Release&quot;</tt>
        <br />
        <tt>null</tt>
      </div>
    ),
    description: (
      <p>
        Title to appear above the select input. The rendered title is also linked to the select
        input with <tt>aria-labelledby</tt> attribute.
      </p>
    ),
  },
];

const releases = [
  {
    release: 'test-tag-1',
    generationDate: '2020-11-10T01:01:30Z',
    url: 'https://test.neonscience.org/local-url',
    productDoi: {
      generationDate: '2020-11-10T01:01:30Z',
      url: 'https://doi.test.datacite.org/10.80606/1wvk-f032',
    },
  },
  {
    release: 'test-tag-3',
    generationDate: '2020-11-20T03:22:30Z',
    url: 'https://test.neonscience.org/local-url',
    productDoi: {
      generationDate: '2020-11-20T03:22:30Z',
      url: 'https://doi.test.datacite.org/10.80606/c93r-j2px',
    },
  },
  {
    release: 'test-tag-2',
    generationDate: '2020-11-15T02:17:30Z',
    url: 'https://test.neonscience.org/local-url',
  },
];

const releasesWithCounts = [
  {
    release: 'test-tag-A',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-10T01:01:30Z',
    dataProductCodes: ['DP1.00001.001', 'DP1.00002.001', 'DP1.00003.001', 'DP1.00004.001'],
    productDoi: {
      generationDate: '2020-11-10T01:01:30Z',
      url: 'https://doi.test.datacite.org/10.80606/1wvk-f032',
    },
  },
  {
    release: 'test-tag-B',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-20T03:22:30Z',
    dataProductCodes: ['DP1.00001.001', 'DP1.00003.001', 'DP1.00004.001'],
  },
  {
    release: 'test-tag-C',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-15T02:17:30Z',
    dataProducts: [
      { productCode: 'DP1.00002.001' },
      { productCode: 'DP1.00004.001' },
    ],
    productDoi: {
      generationDate: '2020-11-15T02:17:30Z',
      url: 'https://doi.test.datacite.org/10.80606/c93r-j2px',
    },
  },
];

const onChange = (newRelease) => {
  console.log(`New release: ${newRelease}`); // eslint-disable-line no-console
};

export default function StyleGuide() {
  const classes = useStyles(Theme);

  return (
    <>

      <DocBlock>
        Use a ReleaseFilter to present a visually consistent means of switching between releases
        on a page or in any workflow where release differentiation is important. Caveats:
      </DocBlock>
      <DocBlock>
        <ul>
          <li>ReleaseFilter is stateless (none of the examples below will persist their state)</li>
          <li>ReleaseFilter is data-product agnostic</li>
        </ul>
      </DocBlock>
      <CodeBlock>
        {`
import ReleaseFilter from 'portal-core-components/lib/components/ReleaseFilter';
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h4" component="h2" gutterBottom>Props</Typography>

      <DocBlock>
        <PropsTable props={propRows} />
      </DocBlock>

      <Divider className={classes.divider} />
      <Typography variant="h4" component="h2" gutterBottom>Usage and Exmaples</Typography>

      <ExampleBlock>
        <ReleaseFilter skeleton />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} onChange={onChange} />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} maxWidth={400} selected="test-tag-2" />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} selected="test-tag-1" showGenerationDate horizontal />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} selected="test-tag-3" showGenerationDate showDoi />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} title={null} excludeNullRelease />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} selected="test-tag-1" showGenerationDate showProductCount showReleaseLink />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter releases={releases} selected="test-tag-1" showGenerationDate showProductCount showReleaseLink releaseLinkDisplayType="Link" />
      </ExampleBlock>
      <CodeBlock>
        {`
const releases = [
  {
    release: 'test-tag-1',
    generationDate: '2020-11-10T01:01:30Z',
    url: 'https://test.neonscience.org/local-url',
    productDoi: {
      generationDate: '2020-11-10T01:01:30Z',
      url: 'https://doi.test.datacite.org/10.80606/1wvk-f032',
    },
  },
  {
    release: 'test-tag-3',
    generationDate: '2020-11-20T03:22:30Z',
    url: 'https://test.neonscience.org/local-url',
    productDoi: {
      generationDate: '2020-11-20T03:22:30Z',
      url: 'https://doi.test.datacite.org/10.80606/c93r-j2px',
    },
  },
  {
    release: 'test-tag-2',
    generationDate: '2020-11-15T02:17:30Z',
    url: 'https://test.neonscience.org/local-url',
  },
];

const onChange = (newRelease) => {
  console.log(\`New release: $\{newRelease}\`);
};

<ReleaseFilter skeleton />
<ReleaseFilter releases={releases} onChange={onChange} />
<ReleaseFilter releases={releases} maxWidth={236} selected="test-tag-2" />
<ReleaseFilter releases={releases} selected="test-tag-1" showGenerationDate />
<ReleaseFilter releases={releases} selected="test-tag-3" showGenerationDate showDoi />
<ReleaseFilter releases={releases} title={null} excludeNullRelease />
<ReleaseFilter releases={releases} showGenerationDate showProductCount showReleaseLink />
<ReleaseFilter releases={releases} selected="test-tag-1" showGenerationDate showProductCount showReleaseLink releaseLinkDisplayType="Link" />
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h4" component="h2" gutterBottom>Including Data Product Counts</Typography>

      <ExampleBlock>
        <ReleaseFilter
          releases={releasesWithCounts}
          showGenerationDate
          showProductCount
          nullReleaseProductCount={42}
        />
      </ExampleBlock>
      <ExampleBlock>
        <ReleaseFilter
          releases={releasesWithCounts}
          selected="test-tag-B"
          showGenerationDate
          showProductCount
          horizontal
        />
      </ExampleBlock>

      <CodeBlock>
        {`
const releasesWithCounts = [
  {
    release: 'test-tag-A',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-10T01:01:30Z',
    dataProductCodes: ['DP1.00001.001', 'DP1.00002.001', 'DP1.00003.001', 'DP1.00004.001'],
    productDoi: {
      generationDate: '2020-11-10T01:01:30Z',
      url: 'https://doi.test.datacite.org/10.80606/1wvk-f032',
    },
  },
  {
    release: 'test-tag-B',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-20T03:22:30Z',
    dataProductCodes: ['DP1.00001.001', 'DP1.00003.001', 'DP1.00004.001'],
  },
  {
    release: 'test-tag-C',
    url: 'https://test.neonscience.org/local-url',
    generationDate: '2020-11-15T02:17:30Z',
    dataProducts: [
      { productCode: 'DP1.00002.001' },
      { productCode: 'DP1.00004.001' },
    ],
    productDoi: {
      generationDate: '2020-11-15T02:17:30Z',
      url: 'https://doi.test.datacite.org/10.80606/c93r-j2px',
    },
  },
];

<ReleaseFilter
  releases={releasesWithCounts}
  showGenerationDate
  showProductCount
  nullReleaseProductCount={42}
/>
<ReleaseFilter
  releases={releasesWithCounts}
  selected="test-tag-B"
  showGenerationDate
  showProductCount
/>
`}
      </CodeBlock>

    </>
  );
}
