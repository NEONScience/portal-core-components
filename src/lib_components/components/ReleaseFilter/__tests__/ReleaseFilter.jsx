import React from 'react';
import renderer from 'react-test-renderer';

import ReleaseFilter from '../ReleaseFilter';

describe('ReleaseFilter', () => {
  test('Renders with no props', () => {
    const tree = renderer
      .create(<ReleaseFilter />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
