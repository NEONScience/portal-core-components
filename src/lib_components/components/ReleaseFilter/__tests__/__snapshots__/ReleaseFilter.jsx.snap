// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseFilter Renders with no props 1`] = `
<div
  style={
    Object {
      "maxWidth": "236px",
      "width": "100%",
    }
  }
>
  <div
    className="makeStyles-titleContainer-2"
  >
    <h3
      className="MuiTypography-root makeStyles-title-1 MuiTypography-h5"
      id="release-filter-label-id1"
    >
      Release
    </h3>
    <button
      aria-describedby={null}
      aria-label={
        <div>
          A data release is a set of data files that is static (unchanging), always available to end users, and citable. See 
          <ForwardRef(WithStyles)
            href="https://www.neonscience.org/data-samples/data-management/data-revisions-releases"
            target="_blank"
          >
            Data Product Revisions and Releases
          </ForwardRef(WithStyles)>
           for more details.
        </div>
      }
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": 4,
        }
      }
      tabIndex={0}
      title={null}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    aria-labelledby="release-filter-label-id1"
    className="MuiInputBase-root MuiOutlinedInput-root makeStyles-selectInput-3 Mui-disabled Mui-disabled MuiInputBase-marginDense MuiOutlinedInput-marginDense"
    data-selenium="release-filter"
    onClick={[Function]}
    style={
      Object {
        "maxWidth": "236px",
      }
    }
  >
    <div
      aria-disabled="true"
      aria-haspopup="listbox"
      aria-labelledby="mui-component-select-release-filter-input-id1"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense Mui-disabled"
      id="mui-component-select-release-filter-input-id1"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={null}
      role="button"
      tabIndex={null}
    >
      Latest and Provisional
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      id="release-filter-input-id1"
      name="release-filter-input-id1"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value="Latest and Provisional"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-14"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
  <div
    className="makeStyles-descriptionContainer-4"
  >
    <span
      className="MuiTypography-root makeStyles-description-6 MuiTypography-caption"
    >
      Data in the latest release in addition to provisional data (not yet in any release)
    </span>
  </div>
</div>
`;
